src/Bicep.Core/Registry/Oci/ArtifactIdParts.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Bicep.Core.Registry.Oci
+{
+    public class ArtifactIdParts : IArtifactIdParts
+    {
+        public ArtifactIdParts(string registry, string repository, string? tag, string? digest)
```

Comment:
Could probably change the class to a `readonly record struct` since it only contains 4 strings.

Reply:
I tried that, esp. since records create their own IEquatable implementation.  Unfortunately, I couldn't verify exactly what string comparison it's using.  It's definitely case-insensitive, but online suggests the default uses culture-aware comparisons, which we wouldn't want.  And apparently I can't override it for records, so I'm leaving as a class and an explicit implementation.

Reply:
Oh good point. I missed the custom comparers part. I guess we can use plain struct instead, but that's a minor improvement we can do later.